platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/HalfDiskHashMap.java

```diff
@@ -411,45 +429,59 @@ public void endWriting() throws IOException {
                 oneTransactionsData.stream().mapToLong(BucketMutation::size).sum());
         // iterate over transaction cache and save it all to file
         if (!oneTransactionsData.isEmpty()) {
-            //  write to files
-            fileCollection.startWriting();
             // for each changed bucket, write the new buckets to file but do not update index yet
-            int oldBucketIndex = -1;
-            for (IntObjectPair<BucketMutation<K>> keyValue :
-                    oneTransactionsData.keyValuesView().toList().sortThis()) {
+            final MutableList<IntObjectPair<BucketMutation<K>>> bucketUpdates =
+                    oneTransactionsData.keyValuesView().toList();
+            final int size = bucketUpdates.size();
+            final ReusableBucketPool<K> bucketPool = bucketSerializer.getBucketPool();
+            final Queue<ReadBucketResult<K>> queue = new ConcurrentLinkedQueue<>();
+            for (int i = 0; i < size; i++) {
+                final IntObjectPair<BucketMutation<K>> keyValue = bucketUpdates.get(i);
                 final int bucketIndex = keyValue.getOne();
-                if (bucketIndex < oldBucketIndex) {
-                    throw new IllegalStateException("Somehow we got our bucket indexes out of order: old="
-                            + oldBucketIndex
-                            + ", new ="
-                            + bucketIndex);
-                }
-                oldBucketIndex = bucketIndex;
                 final BucketMutation<K> bucketMap = keyValue.getTwo();
-                try {
-                    Bucket<K> bucket = fileCollection.readDataItemUsingIndex(bucketIndexToBucketLocation, bucketIndex);
-                    if (bucket == null) {
-                        // create a new bucket
-                        bucket = bucketSerializer.getReusableEmptyBucket();
-                        bucket.setBucketIndex(bucketIndex);
+                flushExecutor.execute(() -> {
+                    try {
```

Comment:
can this try/catch block be a method?
e.g.
```java
flushExecutor.execute(() -> {
  flushBucketImmediate(bucketPool, queue, bucketMap, bucketIndex);
}
```

Reply:
Let me implement this. Will it have any impact on performance, just wondering?

Reply:
Shouldn't have any measurable impact, but worth testing.

Reply:
I couldn't spot any different in my (pretty basic) profiler using CryptoBench benchmarks.